Sprint-1/implement/sum.test.js

```diff
@@ -13,24 +13,80 @@ const sum = require("./sum.js");
 // Given an empty array
 // When passed to the sum function
 // Then it should return 0
-test.todo("given an empty array, returns 0")
+
+test.todo("given an empty array returns 0"), () => {
+    const arr = [];
+
+    const result = sum(arr);
+
+    expect(result).toEqual(0)
+}
 
 // Given an array with just one number
 // When passed to the sum function
 // Then it should return that number
 
+test.todo("given with just one number return the number"), () => {
+    const arr = [56];
+
+    const result = sum(arr);
+
+    expect(result).toEqual(56)
+}
+
+
+
 // Given an array containing negative numbers
 // When passed to the sum function
 // Then it should still return the correct total sum
 
+test.todo("given an array containing negative numbers"), () => {
+    const arr = [-1, -3, -4, -5];
+
+    const result = sum(arr);
+
+    expect(result).toEqual(-13)
+}
+
+
+
 // Given an array with decimal/float numbers
 // When passed to the sum function
 // Then it should return the correct total sum
 
+test.todo("Given an array with decimal/float numbers"), () => {
+    const arr = [1.1, 3.4, 4.5, 8.9];
+
+    const result = sum(arr);
+
+    expect(result).toEqual(17.9)
```

Comment:
Decimal numbers in most programming languages (including JS) are internally represented in "floating point number" format. Floating point arithmetic is not exact. For example, the result of `46.5678 - 46 === 0.5678` is false because `46.5678 - 46` only yield a value that is very close to `0.5678`. Even changing the order in which the program add/subtract numbers can yield different values.

So the following could happen
```
  expect(sum([1.2, 0.6, 0.005])).toEqual(1.805);                // This fail
  expect(sum([1.2, 0.6, 0.005])).toEqual(1.8049999999999997);   // This pass
  expect(sum([0.005, 0.6, 1.2])).toEqual(1.8049999999999997);   // This fail
```

**Can you find a more appropriate way to test a value (that involves decimal number calculations) for equality?**